src/perf/isolates.md

```diff
@@ -0,0 +1,347 @@
+you can move these computations to a helper isolate.
+This allows the underlying runtime environment 
+to run the computation concurrently
+with the main UI isolate's work
+and takes advantage of multi-core devices.
+
+Each isolate has its own memory
+and its own event loop.
+The event loop processes
+events in the order that they're added to an event queue.
+On the main isolate,
+these events can be anything from handling a user tapping in the UI,
+to executing a function,
+to painting a frame on the screen.
+The following figure shows an example event queue
+with 3 events waiting to be processed.
+
+![The main isolate diagram]({{site.url}}/assets/images/docs/development/concurrency/basics-main-isolate.png){:width="50%"}
+
+For smooth rendering,
+Flutter adds a "paint frame" event to the event queue
+60 times per second(for a 60Hz device).
+If these events aren't processed on time,
+the application experiences UI jank,
+or worse,
+become unresponsive altogether.
+
+![Event jank diagram]({{site.url}}/assets/images/docs/development/concurrency/event-jank.png){:width="50%"}
+
+Whenever a process can't be completed in a frame gap,
+or the time between two frames,
+it's a good idea to offload the work to another isolate
+to ensure that the main isolate can produce 60 frames per second.
+When you spawn an isolate in Dart,
+it can process the work concurrently with the main isolate,
+without blocking it.
+
+You can read more about how isolates
+and the event loop work in Dart on
+the [concurrency page][] of the Dart
+documentation.
+
+[concurrency page]: {{site.dart-site}}/language/concurrency
+
+<iframe width="560" height="315" src="{{site.youtube-site}}/embed/vl_AaCgudcY" 
+ frameborder="0" allow="accelerometer; autoplay;encrypted-media; gyroscope; picture-in-picture" 
+ allowfullscreen></iframe>
+
+## Common use cases for isolates
+
+There is only one hard rule for when you should use isolates,
+and that's when large computations are causing your Flutter application
+to experience UI jank.
+This jank happens when there is any computation that takes longer than
+Flutter's frame gap.
+
+![Event jank diagram]({{site.url}}/assets/images/docs/development/concurrency/event-jank.png){:width="50%"}
+
+Any process _could_ take longer to complete,
+depending on the implementation
+and the input data,
+making it impossible to create an exhaustive list of
+when you need to consider using isolates.
+
+That said, isolates are commonly used for the following:
+
+- Reading data from a local database
+- Sending push notifications
+- Parsing and decoding large data files
+- Processing or compressing photos, audio files, and video files
+- Converting audio and video files
+- When you need asynchronous support while using FFI
+- Applying filtering to complex lists or filesystems
+
+## Message passing between isolates
+
+Dart's isolates are an implementation of the [Actor model][].
+They can only communicate with each other by message passing.
```

Comment:
Maybe it's worth mentioning "port"s here?

I think in general in this doc whenever it's using the "passing messages" nomenclature it might be more clear to use Dart nomenclature like "ports".